src/cocoa/toga_cocoa/widgets/tree.py

```diff
@@ -92,38 +96,66 @@ def outlineView_objectValueForTableColumn_byItem_(self, tree, column, item):
         else:
             icon = None
 
-        # Now construct the data object for the cell.
-        # If the datum already exists, reuse and update it.
-        # If an icon is present, create a TogaData object.
-        # Otherwise, just use the string (because a Python string)
-        # is transparently an ObjC object, so it works as a value.
-        obj = item.attrs['values'][col_identifier]
-        if obj is None or isinstance(obj, str):
-            if icon:
-                # Create a TogaData value
-                obj = TogaData.alloc().init()
-                obj.attrs = {
-                    'label': str(value),
-                    'icon': icon,
-                }
-            else:
-                # Create/Update the string value
-                obj = str(value)
-            item.attrs['values'][col_identifier] = obj
+        # creates a NSTableCellView from interface-builder template (does not exist)
+        # or reuses an existing view which is currently not needed for painting
+        # returns None (nil) if both fails
+        tcv = self.makeViewWithIdentifier(at('DataCell'), owner=self)
```

Comment:
Does this identifier need to be unique to *this* widget? As I recall, these cells can't/shouldn't be reused across multiple widget instances. If so, including a reference to `widget.interface.id` may be necessary - every widget should have a unique ID.

Reply:
Oh, I didn't know that. I have switched to a unique identifier now.